### PR TITLE
Feat/addleafletmap

### PR DESCRIPTION
**Pull request by mikemangas**
added following:

1. rendering leaflet map with help of react leaflet package
2. creating a test.json
3. retrieving jsons data and mapping over them, to display the markers (location pointers on the map)

### CODE REVIEW

File: client/src/App.css [hunk 1]
```diff
@@ -1,3 +1,9 @@
+.leaflet-container {
+  height: 100vh;
+  width: 100%;
+}
+
 #mapid {
-  height: 180px;
+  height: 100vh;
+  width: 100%;
 }
```
Comment (johnxgaehner) [inline client/src/App.css:9]: hey Mike,
not sure what you trying to achieve, but if you want to prevent the site from scrolling, you have to put the 100vh in the parent element of your header + main content. if you just want the element to use the full height, you can use 100% height or flex: stretch would work also probably.

File: client/src/components/Header.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ export default function Header() {
   return (
     <header className="header">
       <NavLink to="/">Home</NavLink>
+      <NavLink to="/2">Home2</NavLink>
       <NavLink to="/impressum">Impressum</NavLink>
       <NavLink to="/datenschutz">Datenschutz</NavLink>
       <NavLink to="/wir">Über Uns</NavLink>
```
Comment (johnxgaehner) [inline client/src/components/Header.js:10]: might add a bit semantic structure if you wrap the NavLinks with a nav element.

File: client/src/pages/Home.js [hunk 3]
```diff
@@ -1,3 +1,133 @@
...
+  //     </Marker>
+  //   );
+  // }
+  function handleDies(e) {
+    e.preventDefault();
+    return <MapContainer zoom={1} />;
+  }
+  return (
+    <div>
+      <form onSubmit={handleDies}>
```
Comment (johnxgaehner) [inline client/src/pages/Home.js:95]: what is handleDies? naming could be better.

**Review by johnxgaehner — COMMENTED** [review]
hey mate! added some questions and suggestions which might help.
found a lot of commented code which shouldn't be in the pull request as far as I understood the practice. haha

File: client/src/pages/Home.js [hunk 4]
```diff
@@ -1,3 +1,133 @@
...
+  //   return position === null ? null : (
+  //     <Marker position={position}>
+  //       <Popup>You are here</Popup>
+  //     </Marker>
+  //   );
+  // }
+  function handleDies(e) {
+    e.preventDefault();
+    return <MapContainer zoom={1} />;
+  }
```
Comment (namirsab) [inline client/src/pages/Home.js:92]: This form handler it's confusing. As @johnxgaehner  mentioned, the name is hard to follow. Also, it's returning some JSX. That JSX is never used. Why are you returning it there?

File: client/src/pages/Home.js [hunk 5]
```diff
@@ -1,3 +1,133 @@
...
+  return (
+    <div>
+      <form onSubmit={handleDies}>
+        <input name="plz" id="plz" placeholder="deine PLZ" />
+        <button type="submit">Send</button>
+      </form>
+
+      <MapContainer
+        // ref={mapRef}
```
Comment (namirsab) [inline client/src/pages/Home.js:101]: Commented code 🔥

File: client/src/pages/Home2.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  // return map;
+  //   L.tileLayer('https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}?access_token={accessToken}', {
+  //     attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, Imagery © <a href="https://www.mapbox.com/">Mapbox</a>',
+  //     maxZoom: 18,
+  //     id: 'mapbox/streets-v11',
+  //     tileSize: 512,
+  //     zoomOffset: -1,
+  //     accessToken: 'your.mapbox.access.token'
+  // }).addTo(mymap);
+}
```
Comment (namirsab) [inline client/src/pages/Home2.js:23]: Commented code 🔥

**Review by namirsab — CHANGES_REQUESTED** [review]
Hey @mikemangas 
Lovely second PR, i see you are having fun researching leaflet!
Please take a look to my comments

File: client/src/components/Header.js [hunk 7]
```diff
@@ -4,6 +4,7 @@ export default function Header() {
   return (
     <header className="header">
       <NavLink to="/">Home</NavLink>
+      <NavLink to="/2">Home2</NavLink>
```
[the remark this answers is not in the document]
Comment (mikemangas) [reply client/src/components/Header.js:7]: thank you for the reviews john & namir! this pull request was probably not a good idea to post, since i just was experimenting with stuff! Thank you though for your comments, they really help!

[the remark this answers is not in the document]
Comment (namirsab) [reply client/src/components/Header.js:7]: @mikemangas you can always post PRs for experimenting but its important you describe that in the description, so reviewers know this is not going to be merged but you still want an opinion on something :)

File: client/src/App.css (hunk 1 above)
Comment (mikemangas) [reply client/src/App.css:9]: its a thing from the library.. i have to give to the container a height and a width.

File: client/src/components/Header.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (mikemangas) [reply client/src/components/Header.js:7]: thanks for the info!
